distributed/scheduler.py

```diff
@@ -1590,6 +1596,33 @@ def get_nbytes(self, stream=None, keys=None, summary=True):
 
             return result
 
+    def publish_dataset(self, stream=None, keys=None, data=None, name=None,
+                        client=None):
+        self.client_wants_keys(keys, 'published-%s' % name)
+        self.published_data[name] = {'data': data, 'keys': keys,
+                                     'clients': {client}}
+
+    def unpublish_dataset(self, stream=None, name=None):
+        out = self.published_data.pop(name, {'keys': []})
+        self.client_releases_keys(out['keys'], 'published-%s' % name)
+
+    def list_datasets(self, *args):
+        return list(sorted(self.published_data.keys()))
+
+    def get_dataset(self, stream, name=None, client=None):
+        self.published_data[name]['clients'].add(client)
+        data = self.published_data[name].copy()
+        del data['clients']
+        return data
+
+    def check_published_datasets(self, client):
+        for key, data in self.published_data.copy().items():
+            with ignoring(KeyError):
+                data['clients'].remove(client)
+            if not(data['clients']):
+                del self.published_data[key]
+                self.client_releases_keys(data['keys'], 'published-%s' % key)
```

Comment:
These last two lines are just `unpublish_dataset` yes?  If so then we should probably reuse that function here.  

Alternatively, if datasets no longer disappear when clients disappear then maybe we no longer need to track which clients care about which datasets and this entire function can be removed.